topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorIT.java

```diff
@@ -654,4 +658,44 @@ public void testReconcile(TestContext context) {
     // TODO: What happens if we create and then change labels to the resource predicate isn't matched any more
     //       What then happens if we change labels back?
 
+
+    @Test
+    public void testKafkaTopicCreatedWithOwnerRef(TestContext context) {
+        String topicName = "test-kafka-topic-with-owner-ref";
+
+        // create a CM which will be owner reference for the topic
+        String cmName = "hodor";
+        HashMap<String, String> cmData = new HashMap<>();
+        cmData.put("strimzi", "rulez");
+        kubeClient.configMaps().inNamespace(NAMESPACE).create(new ConfigMapBuilder().withNewMetadata().withName(cmName)
+                .withNamespace(NAMESPACE).endMetadata().withApiVersion("v1").withData(cmData).build());
+        String uid = kubeClient.configMaps().inNamespace(NAMESPACE).withName(cmName).get().getMetadata().getUid();
+
+        ObjectMeta metadata = new ObjectMeta();
+
+        OwnerReference or = new OwnerReferenceBuilder().withName(cmName)
+                .withApiVersion("v1")
+                .withController(false)
+                .withBlockOwnerDeletion(false)
+                .withUid(uid)
+                .withKind("ConfigMap")
+                .build();
+
+        metadata.getOwnerReferences().add(or);
+
+        Map<String, String> annos = new HashMap<>();
+        annos.put("iam", "groot");
+
+        metadata.setAnnotations(annos);
+
+        Topic topic = new Topic.Builder(topicName, 1, (short) 1, emptyMap(), metadata).build();
+        KafkaTopic topicResource = TopicSerialization.toTopicResource(topic, resourcePredicate);
+
+        createKafkaTopicResource(context, topicResource);
+
+        assertEquals(1, operation().inNamespace(NAMESPACE).withName(topicName).get().getMetadata().getOwnerReferences().size());
+        assertEquals(1, operation().inNamespace(NAMESPACE).withName(topicName).get().getMetadata().getAnnotations().size());
+
+    }
```

Comment:
We should also check the modification paths too: Modify the `KafkaTopic` in k8s and check the OR, anno and labels are preserved. Also modify the topic in Kafka and again check that these things are preserved.